packages/core/react/use-chat.ts

```diff
@@ -41,229 +46,323 @@ export type UseChatHelpers = {
   /** An input/textarea-ready onChange handler to control the value of the input */
   handleInputChange: (e: any) => void
   /** Form submission handler to automattically reset input and append a user message  */
-  handleSubmit: (e: React.FormEvent<HTMLFormElement>) => void
+  handleSubmit: (e: React.FormEvent<HTMLFormElement>, functions: ChatCompletionFunctions[]) => void
   /** Whether the API request is in progress */
   isLoading: boolean
 }
 
+export type OpenAIChatRequest = {
+  messages: Message[],
+  functions?: Array<ChatCompletionFunctions>;
+  function_call?: CreateChatCompletionRequestFunctionCall;
+}
+
+export type FunctionCallHandler = (functionCall: ChatCompletionRequestMessageFunctionCall, chatRequest: OpenAIChatRequest) => Promise<OpenAIChatRequest>;
+
 export function useChat({
-  api = '/api/chat',
-  id,
-  initialMessages = [],
-  initialInput = '',
-  sendExtraMessageFields,
-  onResponse,
-  onFinish,
-  onError,
-  headers,
-  body
-}: UseChatOptions = {}): UseChatHelpers {
-  // Generate an unique id for the chat if not provided.
-  const hookId = useId()
-  const chatId = id || hookId
+                          api = '/api/chat',
```

Comment:
Seems like indentation is different here,  and this change added `;`.
There's an script to fix this I guess: `prettier-fix`.

Reply:
Great point, I'll adjust my linting settings and push up a fix later today.